tests/image_utils.py

```diff
@@ -0,0 +1,108 @@
+"""
+Image wrappers used for testing various image types with a common Numpy NDArray
+interface.
+
+An example usage is shown below, where an array is wrapped into an Xarray image,
+modified, unwrapped back to a Numpy array, and compared to another array.
+
+>>> from numpy.testing import assert_array_equal
+>>> array = np.ones((8, 8, 3))
+>>> wrapped = wrap(np.ones((8, 8, 3)), type=xr.Dataset)
+>>> wrapped += 1
+>>> assert_array_equal(unwrap(wrapped), array + 1)
+
+The advantage of this system is that you can easily parameterize over multiple types by
+changing the `type` parameter, without having to modify the test code.
+"""
+
+from dataclasses import dataclass
+from functools import singledispatch
+
+import dask.array as da
+import numpy as np
+import pytest
+import xarray as xr
+from numpy.typing import NDArray
+
+from sknnr_spatial.image._base import ImagePreprocessor
+from sknnr_spatial.image.dataarray import DataArrayPreprocessor
+from sknnr_spatial.image.dataset import DatasetPreprocessor
+from sknnr_spatial.image.ndarray import NDArrayPreprocessor
+from sknnr_spatial.types import ImageType
+
+
+@dataclass
+class TestImageType:
+    """A container for image types used in testing."""
+
+    __test__ = False
```

Comment:
This syntax was new to me!

Reply:
Me too. Probably a sign that I should've chosen a better name for `TestImageType` to avoid confusing pytest, but that was all I could come up with!